### ISSUE TITLE
Allow 2 concurrent debug sessions in the same directory

### ISSUE THREAD

**Issue by kampde** [vim-vdebug/vdebug#97]
I have a problem when I try to debug 2 scripts that reside in the same directory at the same time: one of them executes the other with an `exec` call, so it's another process and the debugger don't automatically jump to the other file.

So I open 2 vims, one with the "parent" script, and the other with the "child" script (the one is executed via `exec`). I put a breakpoint in the exec call in `parent`, then go to `child` and F5 there. Step over in `parent` and then `child` picks the connection.

But this fails because the temp swap files Vdebug creates on the folder (`.Debugger{Stack|Status|Watch}.swp`) when debugging `parent`, and the new debug session in `child` sees these files and prompts with the typical `Abort, Retry, Quit, Edit anyway?` message.

It is possible to put the PID of the process in the name of the swapfiles or some other method to be able to debug those 2 scripts simultaneously?

**Comment by benjifisher** [comment]
I suggested a way to avoid crating swap files in Issue #187, and  later comment there says,

> This fix is added in the v2-integration branch, thanks

I still see the [problematic line of code](https://github.com/joonty/vdebug/blob/master/plugin/python/vdebug/ui/vimui.py#L347) in the `master` branch, and it does not seem to be fixed in what looks like the [corresponding line](https://github.com/joonty/vdebug/blob/v2-integration/pythonx/vdebug/ui/vimui.py#L496) in the `v2-integration` branch.

It is an easy fix:

> it is just a matter of changing `'silent '` to `'silent noswapfile '` (note the trailing spaces).

**Comment by BlackIkeEagle** [comment]
@benjifisher I'm pretty sure the noswapfile was added or was not needed for some reason if it was not added, the relevant commit can be found via #187 so maybe I have to check why it was removed again

**Comment by BlackIkeEagle** [comment]
should be fixed again

**Comment by benjifisher** [comment]
@BlackIkeEagle:

Are you recommending that everyone use the `v2-integration` branch rather than the latest release?  I am trying to understand the policy on when an issue is closed.

I prefer my solution, adding the `noswapfile` modifier along with `silent` when creating the buffer.  After Commit abd4ceb3, the swap file will be created along with the buffer and then deleted when you reset the `'swapfile'` option.  But I am splitting hairs here:  according to `:help 'swapfile'`,

> 	When `'swapfile'` is reset, the swap file for the current buffer is
	immediately deleted.

So your solution should work fine.

**Comment by BlackIkeEagle** [comment]
@benjifisher I would suggest helping out filing bugs on v2-integration, but thatone also needs to be converted to python3, so if you want really stable, use master. But currently I'm trying to get things forward in v2-integration and since this is not a really big issue I don't see the need to bring it in master

**Comment by jimafisk** [comment]
I was having the same problem debugging when one script executes another (using drush with Drupal). Here's how I got this working:

1. Set a breakpoint like normal (In my case a custom module that adds a drush command)
2. Start the debugger like normal in that file with <kbd>F5</kbd>
3. Execute the script (In my case running the custom drush migration command)
4. When the debugger stops in the first script (in my case `~/.composer/vendor/drush/drush/drush`) run `:execute 'python debugger.detach()' | execute 'python debugger.run()'` and then press <kbd>Enter</kbd> to continue when prompted. 
5. If the script stops again (in my case `/path/to/project/vendor/drush/drush/drush.php`), press <kbd>F5</kbd> to continue running and it should get to your breakpoint and stop.

Note I did not need the `'silent noswapfile '` mentioned above. I apologize if this is seen as hijacking this issue, I just hope the extra info can help someone else if they end up here like I did.  

Credit goes to @benjifisher for helping me with this on slack.

 <details>
      <summary>Here's a gif that demonstrates this process</summary>

![debug_drush](https://user-images.githubusercontent.com/5913244/31232231-448e28d8-a9b7-11e7-944d-aa804b224b9d.gif)

</details>

**Comment by benjifisher** [comment]
@jimafisk I am glad that suggestion worked.  You do not need to worry about the swap file, since you are not running two instances of Vdebug.

To clarify:  the key command

```vim
:execute 'python debugger.detach()' | execute 'python debugger.run()'
```

has the effect of detaching the debugger (like `F7`) and then immediately starting a new session (like `F5`).  I figure this is more reliable than trying to hit the two keys manually.

**Comment by jimafisk** [comment]
Ahh yes, good point. I omitted <kbd>F7</kbd> to <kbd>F5</kbd> alternative because I wasn't able to get that to work initially. If you do that in very quick succession it does indeed work!

**Comment by BlackIkeEagle** [comment]
@jimafisk in the v2-integration branch it should work by hitting F7 but I think there is still some bug there, have to check
